### PR TITLE
fix(graphql-auth-transformer): verify multiple static group auth rules

### PR DESCRIPTION
**Pull request by SwaySway**
*Issue #, if available:*
re #2241
*Description of changes:*
ensure multiple static group rules are verified per each field

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/graphql-auth-transformer/src/resources.ts [hunk 1]
```diff
@@ -813,11 +816,13 @@ identityClaim: "${rule.identityField || rule.identityClaim || DEFAULT_IDENTITY_F
     // A = IsStaticallyAuthed
     // B = AuthConditionIsNotNull
     // ! (A OR B) == (!A AND !B)
-    public throwIfNotStaticGroupAuthorizedOrAuthConditionIsEmpty(): Expression {
+    public throwIfNotStaticGroupAuthorizedOrAuthConditionIsEmpty(fieldName?: string): Expression {
+        const staticGroupAuthorizedVariable = fieldName ? `${fieldName}_${ResourceConstants.SNIPPETS.IsStaticGroupAuthorizedVariable}` :
```
Comment (yuth) [inline packages/graphql-auth-transformer/src/resources.ts]: can we move this to a helper function

Comment (SwaySway) [reply packages/graphql-auth-transformer/src/resources.ts]: added a helper function in latest commit

File: packages/graphql-auth-transformer/src/resources.ts [hunk 2]
```diff
@@ -1022,6 +1018,11 @@ identityClaim: "${rule.identityField || rule.identityClaim || DEFAULT_IDENTITY_F
         return block("Determine request authentication mode", expressions);
     }
 
+    public setStaticAuthVariable(field: FieldDefinitionNode): string {
```
Comment (yuth) [inline packages/graphql-auth-transformer/src/resources.ts]: this should be called `getStaticAuthVariableName` as we are not setting the variable itself in this method

File: packages/graphql-auth-transformer/src/ModelAuthTransformer.ts [hunk 3]
```diff
@@ -1287,8 +1291,10 @@ All @auth directives used on field definitions are performed when the field is r
                 )
 
                 const collectAuthCondition = this.resources.collectAuthCondition()
+                const staticGroupAuthorizedVariable = field ? `${field.name.value}_${ResourceConstants.SNIPPETS.IsStaticGroupAuthorizedVariable}` :
```
Comment (yuth) [inline packages/graphql-auth-transformer/src/ModelAuthTransformer.ts]: leftover? can we use the helper method here?

Comment (SwaySway) [reply packages/graphql-auth-transformer/src/ModelAuthTransformer.ts]: removed leftover

File: packages/graphql-auth-transformer/src/resources.ts (hunk 2 above)
Comment (SwaySway) [reply packages/graphql-auth-transformer/src/resources.ts]: changed to set

**Pull request merged**